doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
+These branches should have the prefix `features/` to indicate they are either
+not complete or have not been scheduled for the next minor release.
 
 Every issue, no matter how small, gets its own branch while under development. Issue branch names are always prefixed with `issues/`, followed by the issue number in the GitHub issue tracker, followed by one or two dash-separated keywords for the issue.
 
 For example: `issues/#276-rdfxml-sax` is the branch for a fix for GitHub issue https://github.com/eclipse/rdf4j/issues/276[#276], which has to do with RDF/XML and the SAX parser.
 
-Once a feature/fix is complete, tested, and reviewed (via a Pull Request), it is merged into master, and the issue branch is closed.
+Once a issue is complete and tested, a Pull Request should be created for
+peer review.
+
+=== Patch Requests
+
+If the change is a bug fix, contains no new features, and
+does not change any public or protected APIs (class or method signatures in the
+core module, i.e. code included in the release archive) then
+
+1. the Pull Request should target the _master_ branch. Once the Pull Request is created
+2. it should be reviewed and any modifications can be made to the issue branch as recommended.
+3. Once a Pull Request has been approved it can be
+4. scheduled for a patch release (if not already) by updating the milestone label.
+5. Only when a Pull Request is approved *and* scheduled for the next patch release it can be merged
+into the _master_ branch then
+6. the issue branch can be closed.
+7. After a PR has been merged into the _master_ branch, the _master_ branch should
+then be merged into the _develop_ branch and any conflicts (such as due to new
+features) should be resolved.
+This can be done from the command line from a clean checkout as follows:
+
+    git checkout develop
+    git pull origin master
+    git push origin develop
+
+=== Feature Requests
+
+Pull Requests that don't qualify for a patch release should
+
+1. target the _develop_ branch or a relevant feature branch.
+2. Once a Pull Request has been approved it can be
+3. scheduled for a major/minor release by updating the milestone label.
+4. Only when a Pull Request is approved *and* scheduled for the next minor release it can be merged
+into the _develop_ branch, only then can
+5. the issue/feature branch can be closed.
+
+Minor and major releases require a formal https://www.eclipse.org/projects/handbook/#release-review[release review], and because this is the case, they need to be planned well in advance, and the project lead needs to manage what can go into each release, and prepare necessary documentation (both technical and legal) for review.
+For this reason approved Pull Requests may stay open for some time until a
+release plan that incorporates these changes is in place.
```

Comment:
The only way to keep a Pull Request "open" is to not merge the changes. Otherwise GitHub will close it automatically. I am not sure I understand the rationale behind why the Pull Request state relates to the legal review, given legal reviews only apply to dependency changes (I thought).